### ISSUE TITLE
Cannot compile with i686-w64-mingw32 v10.2

### ISSUE THREAD

**Issue by yatli** [RamonUnch/AltDrag#38]
```
 ┌─┐ F:\git\altdrag                                                                                                                                                                                                                                                              main ◇ ┆ Yatao
 └─╜ ╰▷ .\mk.bat

F:\git\altdrag>gcc -o AltDrag.exe altdrag.c altdragr.o -Os -fno-stack-check -fno-stack-protector -mno-stack-arg-probe -march=i386 -mtune=i686 -mpreferred-stack-boundary=2 -foptimize-strlen -nostdlib -lkernel32 -luser32 -lmsvcrt -flto -Wl,-dynamicbase,-nxcompat -Wp,-D_FORTIFY_SOURCE=2 -s -
Wall -Wformat-security -Wstrict-overflow -Wsign-compare -Wclobbered -Wempty-body -Wignored-qualifiers -Wuninitialized -Wtype-limits -Woverride-init -Wduplicated-cond  -Wduplicated-branches -Wlogical-op  -Wnull-dereference -lcomctl32 -mwindows -ladvapi32 -lshell32 -e_unfuckWinMain@0
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x2e): undefined reference to `__mingw_vsnwprintf'
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x830): undefined reference to `__chk_fail'
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x1725): undefined reference to `__chk_fail'
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x183a): undefined reference to `__chk_fail'
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x20cf): undefined reference to `__chk_fail'
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x2546): undefined reference to `__mingw_vswscanf'
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x2a14): undefined reference to `__chk_fail'
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\AltDrag.exe.4nPatO.ltrans0.ltrans.o:<artificial>:(.text+0x321e): undefined reference to `__chk_fail'
collect2.exe: error: ld returned 1 exit status

F:\git\altdrag>gcc -o hooks.dll hooks.c hooksr.o -Os -fno-stack-check -fno-stack-protector -mno-stack-arg-probe -march=i386 -mtune=i686 -mpreferred-stack-boundary=2 -foptimize-strlen -nostdlib -lkernel32 -luser32 -lmsvcrt -flto -Wl,-dynamicbase,-nxcompat -Wp,-D_FORTIFY_SOURCE=2 -s -Wall -
Wformat-security -Wstrict-overflow -Wsign-compare -Wclobbered -Wempty-body -Wignored-qualifiers -Wuninitialized -Wtype-limits -Woverride-init -Wduplicated-cond  -Wduplicated-branches -Wlogical-op  -Wnull-dereference -mdll -lgdi32 -e_DllMain@12
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:\Users\Yatao\AppData\Local\Temp\hooks.dll.bTKd7j.ltrans0.ltrans.o:<artificial>:(.text+0xca): undefined reference to `__mingw_vswscanf'
collect2.exe: error: ld returned 1 exit status
The system cannot find the file AltDrag.exe.
```

Output of `gcc --version -v`:
```
 └─╜ ╰▷ gcc --version -v
Using built-in specs.
COLLECT_AS_OPTIONS='--version'
COLLECT_GCC=C:\ProgramData\chocolatey\lib\mingw\tools\install\mingw32\bin\gcc.exe
COLLECT_LTO_WRAPPER=c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/lto-wrapper.exe
gcc.exe (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


Target: i686-w64-mingw32
Configured with: ../gcc-10.2.0/configure --build=i686-w64-mingw32 --target=i686-w64-mingw32 --disable-multilib --disable-64bit --prefix=/mingw32 --with-sysroot=/mingw32 --disable-shared --enable-static --disable-nls --enable-version-specific-runtime-libs --disable-win32-registry --without
-dwarf2 --enable-sjlj-exceptions --enable-fully-dynamic-string --enable-languages=c,d,ada,lto,c++,objc,obj-c++,fortran --enable-libgomp --enable-lto --enable-libssp -enable-gnattools --disable-bootstrap --with-gcc --with-gnu-as --with-gnu-ld --with-stabs --enable-interwork --with-mpfr-inc
lude=/home/betta/gcc-build/../gcc-10.2.0/mpfr/src --with-mpfr-lib=/home/betta/gcc-build/mpfr/src/.libs --enable-objc-gc --with-target-bdw-gc=/mingw32 --disable-dependency-tracking
Thread model: win32
Supported LTO compression algorithms: zlib
gcc version 10.2.0 (GCC)
COLLECT_GCC_OPTIONS='--version' '-v' '-mtune=generic' '-march=pentiumpro'
 c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/cc1.exe -quiet -v -iprefix c:\programdata\chocolatey\lib\mingw\tools\install\mingw32\bin\../lib/gcc/i686-w64-mingw32/10.2.0/ -isysroot c:\programdata\chocolatey\lib\mingw\tools\install\mi
ngw32\bin\../../mingw32 -U_REENTRANT help-dummy -quiet -dumpbase help-dummy -mtune=generic -march=pentiumpro -auxbase help-dummy -version --version -o C:\Users\Yatao\AppData\Local\Temp\ccT7odBY.s
GNU C17 (GCC) version 10.2.0 (i686-w64-mingw32)
        compiled by GNU C version 10.2.0, GMP version 6.2.0, MPFR version 4.1.0, MPC version 1.2.0, isl version isl-0.22.1-GMP

GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
COLLECT_GCC_OPTIONS='--version' '-v' '-mtune=generic' '-march=pentiumpro'
 c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/as.exe -v --version -o C:\Users\Yatao\AppData\Local\Temp\ccwymplS.o C:\Users\Yatao\AppData\Local\Temp\ccT7odBY.s
GNU assembler version 2.35.1 (i686-w64-mingw32) using BFD version (GNU Binutils) 2.35.1
GNU assembler (GNU Binutils) 2.35.1
Copyright (C) 2020 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or later.
This program has absolutely no warranty.
This assembler was configured for a target of `i686-w64-mingw32'.
COMPILER_PATH=c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/;c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/;c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.
0/../../../../i686-w64-mingw32/bin/
LIBRARY_PATH=c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/;c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/;c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../.
./../i686-w64-mingw32/lib/../lib/;c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../lib/;c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/lib/;c:/program
data/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../
COLLECT_GCC_OPTIONS='--version' '-v' '-mtune=generic' '-march=pentiumpro'
 c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/collect2.exe -plugin c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/liblto_plugin-0.dll -plugin-opt=c:/programdata/chocolatey/lib/min
gw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/lto-wrapper.exe -plugin-opt=-fresolution=C:\Users\Yatao\AppData\Local\Temp\cc92rALS.res -plugin-opt=-pass-through=-lmingw32 -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lmoldname -plugin-opt=-pass-through=-l
mingwex -plugin-opt=-pass-through=-lmsvcrt -plugin-opt=-pass-through=-ladvapi32 -plugin-opt=-pass-through=-lshell32 -plugin-opt=-pass-through=-luser32 -plugin-opt=-pass-through=-lkernel32 -plugin-opt=-pass-through=-lmingw32 -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lmoldn
ame -plugin-opt=-pass-through=-lmingwex -plugin-opt=-pass-through=-lmsvcrt --sysroot=c:\programdata\chocolatey\lib\mingw\tools\install\mingw32\bin\../../mingw32 -m i386pe -Bdynamic --version c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/.
./../../../i686-w64-mingw32/lib/../lib/crt2.o c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/crtbegin.o -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0 -Lc:/programdata/chocolatey/lib/ming
w/tools/install/mingw32/bin/../lib/gcc -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/lib/../lib -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../
lib -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/lib -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../.. C:\Users\Yatao\AppData\Local\Temp\ccwymplS.o
-lmingw32 -lgcc -lmoldname -lmingwex -lmsvcrt -ladvapi32 -lshell32 -luser32 -lkernel32 -lmingw32 -lgcc -lmoldname -lmingwex -lmsvcrt c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/crtend.o
collect2 version 10.2.0
c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/bin/ld.exe -plugin c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/liblto_plugin-0.dll -plugin-opt=c:/programd
ata/chocolatey/lib/mingw/tools/install/mingw32/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/lto-wrapper.exe -plugin-opt=-fresolution=C:\Users\Yatao\AppData\Local\Temp\cc92rALS.res -plugin-opt=-pass-through=-lmingw32 -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lmoldname -plugi
n-opt=-pass-through=-lmingwex -plugin-opt=-pass-through=-lmsvcrt -plugin-opt=-pass-through=-ladvapi32 -plugin-opt=-pass-through=-lshell32 -plugin-opt=-pass-through=-luser32 -plugin-opt=-pass-through=-lkernel32 -plugin-opt=-pass-through=-lmingw32 -plugin-opt=-pass-through=-lgcc -plugin-opt
=-pass-through=-lmoldname -plugin-opt=-pass-through=-lmingwex -plugin-opt=-pass-through=-lmsvcrt --sysroot=c:\programdata\chocolatey\lib\mingw\tools\install\mingw32\bin\../../mingw32 -m i386pe -Bdynamic --version c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i68
6-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/lib/../lib/crt2.o c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/crtbegin.o -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0 -Lc:/programda
ta/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/lib/../lib -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw
32/10.2.0/../../../../lib -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../../../i686-w64-mingw32/lib -Lc:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/../../.. C:\Users\Yatao\AppData\
Local\Temp\ccwymplS.o -lmingw32 -lgcc -lmoldname -lmingwex -lmsvcrt -ladvapi32 -lshell32 -luser32 -lkernel32 -lmingw32 -lgcc -lmoldname -lmingwex -lmsvcrt c:/programdata/chocolatey/lib/mingw/tools/install/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.2.0/crtend.o
GNU ld (GNU Binutils) 2.35.1
Copyright (C) 2020 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
COLLECT_GCC_OPTIONS='--version' '-v' '-mtune=generic' '-march=pentiumpro'
```

**Comment by yatli** [comment]
`__mingw_vsnwprintf` can be resolved by removing `-nostdlib`

The other errors are just mingw32 configuration problems with gcc itself. The workaround is to remove `-Wp,-D_FORTIFY_SOURCE=2 `

**Comment by yatli** [comment]
`unfuckWinMain` works fine without `-nostdlib` :D

**Comment by RamonUnch** [comment]
I will see this, I was using GCC8.1.
I disabled stdlib because I experienced problems with it in the past with mingw, and in general for Win32 programs I really see no point in this stdlib that is made for GNU compatibility.
the unfuckWinMain is just the entrypoint hence the `-e_unfuckWinMain` and calls `WinMain()`. if you remove the `-nosdlib` `_WinMain@16` will be resolved by the linker and you do not need the to precise the entry point with `-e`
